### DISCUSSION TITLE
Inline elements

### DISCUSSION THREAD

**Discussion by DanMan32** [Hieromon/AutoConnect#384] [How can I?] [answered]
How can one place elements next to each other?
I've read several ways to do it in HTML where elements can be grouped together that you want inline, but not sure how to code that using Autoconnect elements, especially if one wants several groups on a page

**Comment by Hieromon** [comment, marked as answer]
A placing way choice for horizontally arranging the AutoConnect elements must match the line feed characteristics of the HTML generated by AutoConnect. AutoConnect will not actively be involved in the layout of custom Web pages generated from AutoConnectElements. However, each element has an attribute to arrange placement on a custom web page horizontally or vertically.

All elements of AutoConnectElement have attributes to explicitly specify the placement direction. All elements of AutoConnectElement have an attribute to explicitly specify the placement direction. It is described in the [documentation](https://hieromon.github.io/AutoConnect/acelements.html#post) as the `post`. Also, how to specify `post` in JSON is explained [here](https://hieromon.github.io/AutoConnect/acjson.html#posterior).

Custom web pages are converted to HTML as unstyled DIV blocks, so you can define nested DIV blocks to place AutoConnect elements horizontally. Therefore the DIV block that instructs the horizontal arrangement using inline-block can be implemented by [**ACElement**](https://hieromon.github.io/AutoConnect/acjson.html#json-object-for-autoconnectelements) as follows:

```cpp
#include <Arduino.h>
#include <WiFi.h>
#include <WebServer.h>
#include <AutoConnect.h>

const char ELEMENTS[] PROGMEM = R"(
{
  "uri": "/elements",
  "title": "Elements",
  "menu": true,
  "element": [
    {
      "name": "header",
      "type": "ACText",
      "value": "Horizontal Placement"
    },
    {
      "name": "sep",
      "type": "ACElement",
      "value": "<hr>"
    },
    {
      "name": "check",
      "type": "ACCheckBox",
      "value": "check",
      "label": "Check",
      "labelPosition": "infront",
      "arrange": "horizontal",
      "posterior": "none"
    },
    {
      "name": "d1o",
      "type": "ACElement",
      "value": "<div style='display:inline-block;vertical-align:middle'>"
    },
    {
      "name": "radio",
      "type": "ACRadio",
      "value": [
        "Button-1",
        "Button-2",
        "Button-3"
      ],
      "label": "Radio",
      "arrange": "vertical",
      "checked": 1,
      "posterior": "none"
    },
    {
      "name": "d1c",
      "type": "ACElement",
      "value": "</div>"
    },
    {
      "name": "d2o",
      "type": "ACElement",
      "value": "<div style='display:inline-block;vertical-align:middle'>"
    },
    {
      "name": "select",
      "type": "ACSelect",
      "option": [
        "Option-1",
        "Option-2",
        "Option-3"
      ],
      "label": "Select",
      "selected": 2,
      "posterior": "none"
    },
    {
      "name": "d2c",
      "type": "ACElement",
      "value": "</div>"
    }
  ]
}
)";

AutoConnect portal;

void setup() {
  portal.load(ELEMENTS);
  portal.begin();
}

void loop() {
  portal.handleClient();
}
```

![horitontal](https://user-images.githubusercontent.com/12591771/127991981-263dcea2-d088-4fb2-97de-e03ad41f3f7a.png)
